src/reflect/scala/reflect/internal/tpe/TypeMaps.scala

```diff
@@ -698,18 +703,30 @@ private[internal] trait TypeMaps {
     // OPT this check was 2-3% of some profiles, demoted to -Xdev
     if (isDeveloper) assert(sameLength(from, to), "Unsound substitution from "+ from +" to "+ to)
 
+    private[this] var fromHasTermSymbol = false
+    private[this] var fromMin = Int.MaxValue
```

Comment:
Funny, I was re-reading about bloom filters while making this change. I guess this is kindred (we can cheaply determine non-existence in a set without iterating the full set), but OTOH we there isn't any superimposition of information.

I did learn about an [Zatocoding](https://en.wikipedia.org/wiki/Superimposed_code) in my travels, sort of an analog bloom filter for librarians.